libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -309,259 +338,137 @@ BES_Base::ProcRetVal BES_Select::processIdleSceneRightMouseButtonReleased(
-          tl::optional<Uuid> footprintUuid;
-          if (devInst.getLibPackage().getFootprints().contains(uuid)) {
-            // change footprint
-            footprintUuid = uuid;
-          } else {
-            // change device
-            deviceUuid = uuid;
-          }
-          CmdReplaceDevice* cmd = new CmdReplaceDevice(
-              mWorkspace, *board, devInst, deviceUuid, footprintUuid);
-          mUndoStack.execCmd(cmd);
-        } catch (Exception& e) {
-          QMessageBox::critical(&mEditor, tr("Error"), e.getMsg());
-        }
-      } else if (action == aProperties) {
-        openDevicePropertiesDialog(devInst);
-      }
-      return ForceStayInState;
+      addActionProperties(menu, board, selectedItem);
+      break;
     }
 
     case BI_Base::Type_t::NetLine: {
-      BI_NetLine* netline = dynamic_cast<BI_NetLine*>(items.first());
+      BI_NetLine* netline = dynamic_cast<BI_NetLine*>(selectedItem);
       Q_ASSERT(netline);
 
-      // build the context menu
-      QAction* aRemoveSegment = menu.addAction(
-          QIcon(":/img/actions/delete.png"), "Remove Trace Segment");
-      QAction* aRemoveTrace = menu.addAction(QIcon(":/img/actions/minus.png"),
-                                             "Remove Whole Trace");
+      addActionDelete(menu, tr("Remove Trace Segment"));
+      addActionDeleteAll(menu, &netline->getNetSegment());
       menu.addSeparator();
-      QAction* aSelectTrace = menu.addAction(
-          QIcon(":/img/actions/bookmark.png"), tr("Select Whole Trace"));
+      addActionSelectAll(menu, &netline->getNetSegment());
       menu.addSeparator();
-      QAction* aMeasureSelection =
-          menu.addAction(QIcon(":/img/actions/ruler.png"),
-                         tr("Measure Selected Segments Length"));
-
-      // execute the context menu
-      QAction* action = menu.exec(mouseEvent->screenPos());
-      if (action == nullptr) {
-        // aborted --> nothing to do
-      } else if (action == aRemoveSegment) {
-        removeSelectedItems();
-      } else if (action == aRemoveTrace) {
-        netline->getNetSegment().setSelected(true);
-        removeSelectedItems();
-      } else if (action == aSelectTrace) {
-        netline->getNetSegment().setSelected(true);
-      } else if (action == aMeasureSelection) {
-        measureSelectedItems(*netline);
-      }
-      return ForceStayInState;
+      addActionMeasure(menu, netline);
+      break;
+    }
+
+    case BI_Base::Type_t::NetPoint: {
+      BI_NetPoint* netpoint = dynamic_cast<BI_NetPoint*>(selectedItem);
+      Q_ASSERT(netpoint);
+
+      addActionDeleteAll(menu, &netpoint->getNetSegment());
+      menu.addSeparator();
+      addActionSelectAll(menu, &netpoint->getNetSegment());
+      menu.addSeparator();
+      addActionSnap(menu, netpoint->getPosition(), board, selectedItem);
+      menu.addSeparator();
+      addActionMeasure(menu, *netpoint->getNetLines().begin());
```

Comment:
I agree it is unlikely, but at least theoretically it's possible that a netpoint has no netlines connected to it, so for safety reasons the netlines iterator should not be blindly dereferenced:

```suggestion
      if (!netpoint->getNetLines().isEmpty()) {
        menu.addSeparator();
        addActionMeasure(menu, *netpoint->getNetLines().begin());
      }
```